pkg/mover/chart.go

```diff
@@ -131,48 +128,56 @@ type ChartMoveRequest struct {
 	Target Target
 }
 
-// ChartMover represents a Helm Chart moving relocation. It's initialization must be done view NewChartMover
-type ChartMover struct {
-	chartDestination        string
-	imageChanges            []*internal.ImageChange
-	chartChanges            []*internal.RewriteAction
-	sourceContainerRegistry internal.ContainerRegistryInterface
-	targetContainerRegistry internal.ContainerRegistryInterface
-	targetOfflineTarPath    string
-	chart                   *chart.Chart
-	logger                  Logger
-	retries                 uint
+// ChartData is the part of the Chart mover data that should be saved for a move to complete
+// without any network access to the original chart resources
+type ChartData struct {
+	chart        *chart.Chart
+	imageChanges []*internal.ImageChange
```

Comment:
I am confused by this new struct, isn't an image change dependent on the destination?

Reply:
in fact it looks like an overkill to just being passed to saveIntermediateBundle, shouldn't we just pass there the images to save?

Reply:
Yes, but turns our we do not have the image lists, only the image changes wrapping all. So I need these to save the state.

Reply:
I am not only using the image but also the image reference:
~~~go
func packImages(w io.Writer, imageChanges []*internal.ImageChange, logger Logger) error {
	refToImage := map[name.Reference]v1.Image{}
	for _, change := range imageChanges {
		if previousImage, ok := refToImage[change.ImageReference]; ok {
			if err := deduplicateByDigest(change.ImageReference.Name(), change.Image, previousImage); err != nil {
				return err
			}
			continue
		}
		refToImage[change.ImageReference] = change.Image
		logger.Printf("Processing image %s\n", change.ImageReference.Name())
	}
~~~
So a list of images will not do, I need images AND references.

In fact I suspect that is the same reason the code does hold that list of changes in the structure. from this:
~~~go
type ImageChange struct {
	Pattern            *ImageTemplate
	ImageReference     name.Reference
	RewrittenReference name.Reference
	Image              v1.Image
	Digest             string
	AlreadyPushed      bool
}
~~~

The only things we need to store persistently from `newChartMove()` to `Move()` are:
~~~go
type ChartImage struct {
	Pattern            *ImageTemplate
	ImageReference     name.Reference
	Image              v1.Image
}
~~~

But all this is bike-shedding, it should be fixed in a separate PR IMHO.

Reply:
In that case I'd not apply this structural change in this patch.